mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
-                                 "disk": 40}}
-
-    def extra_specs_json(self):
+    def flavor_by_id(self, flavor_id):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a memory flavor.
+        Retrieve a :obj:`Flavor` object by its ID.
         """
-        return {
-            "class": "io1",
-            "policy_class": "io_flavor"
-        }
-
-
-class RackspaceMemoryFlavor(Flavor):
-    """
-    A Rackspace memory flavor object representation
-    """
-    flavors = {"15 GB Memory v1": {"id": "memory1-15", "ram": 15360, "vcpus": 2, "rxtx_factor": 625,
-                                   "disk": 0},
-               "30 GB Memory v1": {"id": "memory1-30", "ram": 30720, "vcpus": 4, "rxtx_factor": 1250,
-                                   "disk": 0},
-               "60 GB Memory v1": {"id": "memory1-60", "ram": 61440, "vcpus": 8, "rxtx_factor": 2500,
-                                   "disk": 0},
-               "120 GB Memory v1": {"id": "memory1-120", "ram": 122880, "vcpus": 16, "rxtx_factor": 5000,
-                                    "disk": 0},
-               "240 GB Memory v1": {"id": "memory1-240", "ram": 245760, "vcpus": 32,
-                                    "rxtx_factor": 10000, "disk": 0}}
+        if len(self.flavors_store) < 1:
+            self.create_flavors_list()
+        for flavor in self.flavors_store:
+            if flavor.flavor_id == flavor_id:
+                return flavor
 
-    def extra_specs_json(self):
+    def create_flavors_list(self):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a flavor.
+        Generates the data for each flavor in each flavor class
         """
-        return {
-            "class": "memory1",
-            "policy_class": "memory_flavor"
-        }
+        rackspace_flavors = [RackspaceStandardFlavor, RackspaceComputeFlavor,
+                             RackspacePerformance1Flavor, RackspaceOnMetalFlavor,
+                             RackspacePerformance2Flavor, RackspaceMemoryFlavor,
+                             RackspaceIOFlavor, RackspaceGeneralFlavor]
 
+        if len(self.flavors_store) < 1:
+            for flavor_class in rackspace_flavors:
+                for flavor, flavor_spec in flavor_class.flavors.iteritems():
+                    flavor_name = flavor
+                    flavor_id = flavor_spec['id']
+                    ram = flavor_spec['ram']
+                    vcpus = flavor_spec['vcpus']
+                    network = flavor_spec['rxtx_factor']
+                    disk = flavor_spec['disk']
+                    tenant_id = self.tenant_id
+                    flavor = flavor_class(flavor_id=flavor_id, tenant_id=tenant_id,
+                                          name=flavor_name, ram=ram, vcpus=vcpus,
+                                          rxtx=network, disk=disk)
```

Comment:
non-blocking: naming everything twice just to shove it into keyword args seems a bit repetitive. I would have had no idea what `rxtx_factor` is, but you have to turn around and call it `rxtx` again.

Anyway, I'm down with it - no need to change this if you don't want.